tests/test/object_safe.rs

```diff
@@ -0,0 +1,29 @@
+use super::*;
+
+/// The current implementation of RustIrDatabase::is_object_safe only checks for
+/// the #[object_safe] flag. This test will start failing once it performs a
+/// proper test because Foo should be object safe.
+#[test]
+fn by_default_not_object_safe() {
```

Comment:
I don't think this is super helpful. For testing the flag, let's just add a `trait Bar {}` to the test below and a `goal { ObjectSafe(Bar) } yields { "No possible solution" }`